lib/ruby_terraform/command_line/options.rb

```diff
@@ -0,0 +1,44 @@
+require_relative 'option'
+require_relative 'flag'
+require_relative 'boolean_option'
+
+module RubyTerraform
+  module CommandLine
+    class Options < Array
```

Comment:
I'm not a fan of extending Ruby's data structures. I think what we are dealing with here is an `OptionSet` or something, specifically representing command line options, which doesn't necessarily follow the semantics of an `Array`. I would say an `OptionSet` _has an_ `Array` (although we could use a different data structure in future if it was found to make more sense) rather than that an `OptionSet` _is an_ `Array`. I could be convinced that an `OptionSet` is an `Enumerable` but even then, only if it really made sense.